op-supervisor/supervisor/backend/db/query.go

```diff
@@ -480,3 +480,46 @@ func (db *ChainsDB) IteratorStartingAt(chain eth.ChainID, sealedNum uint64, logI
 	}
 	return logDB.IteratorStartingAt(sealedNum, logIndex)
 }
+
+// FindFirstBlockReferencingLogs returns the first block in the given foreign chain that contains messages
+// depending on logs from the source block or its descendants.
+func (db *ChainsDB) FindFirstBlockReferencingLogs(sourceBlock eth.BlockRef, sourceChainIndex types.ChainIndex, foreignChainID eth.ChainID) (eth.BlockRef, bool, error) {
```

Comment:
nit: I would recommend "destination" over "foreign" because technically the chain can reference itself, in which case it isn't foreign.